armi/reactor/reactors.py

```diff
@@ -2280,17 +2286,21 @@ def processLoading(self, cs, dbLoad: bool = False):
                 for a in self.getAssemblies(includeAll=True):
                     a.makeAxialSnapList(self.refAssem)
 
-            if not cs["inputHeightsConsideredHot"]:
-                runLog.header(
-                    "=========== Axially expanding all (except control) assemblies from Tinput to Thot ==========="
-                )
-                axialExpChngr = AxialExpansionChanger(cs["detailedAxialExpansion"])
-                for a in self.getAssemblies():
-                    if not a.hasFlags(Flags.CONTROL):
-                        axialExpChngr.setAssembly(a)
-                        axialExpChngr.expansionData.computeThermalExpansionFactors()
-                        axialExpChngr.axiallyExpandAssembly(thermal=True)
-                axialExpChngr.manageCoreMesh(self.parent)
+            for a in self.getAssemblies(includeAll=True):
+                assemsToAxiallyExpand.append(a)
+
+        if not cs["inputHeightsConsideredHot"]:
+            runLog.header(
+                f"=========== Axially expanding {aExpStr} assemblies (except control) from Tinput to Thot ==========="
+            )
+            axialExpChngr = AxialExpansionChanger(cs["detailedAxialExpansion"])
+            for a in assemsToAxiallyExpand:
+                if not a.hasFlags(Flags.CONTROL):
+                    axialExpChngr.setAssembly(a)
+                    axialExpChngr.expansionData.computeThermalExpansionFactors()
+                    axialExpChngr.axiallyExpandAssembly(thermal=True)
+            axialExpChngr.manageCoreMesh(self.parent)
```

Comment:
I dont think we want to call manager core mesh when its only doing blueprints, right? They should just stay as they were in the database?

What do you think about moving this code back inside the else statement of IsDB and doing a separate expansions of the blueprints after the if/else?

I also think the blueprints need setBlockMeshCalled on them, and this will never be called since manageCoreMesh only calls it on 
a in r.core.getAssemblies():

We dont want the blueprints to use the core mesh anyway though, since we dont want blueprints to have the burnup mesh the core has, just the thermal mesh.

Reply:
Addressed. https://github.com/terrapower/armi/pull/823/commits/15499a11f58af5a0803e4645e818c488da9b0f4b